apps/wallet-mobile/src/Receive/AddressDetail.tsx

```diff
@@ -1,25 +1,322 @@
+          style={{ height: '100%' }}
+          data={data}
+          keyExtractor={(item, index) => index.toString()}
+          pagingEnabled
+          onScroll={onPageChange}
+          snapToInterval={itemsPerPage * SCREEN_WIDTH}
+          decelerationRate="fast"
+          renderItem={({ item }) => {
+            if (item.cardType === 'QRCode') {
+              return <ShareQRCodeCard title={item.title} address={item.address} />;
+            } else if (item.cardType === 'Details') {
+              return <ShareDetailsCard address={item.address} stakingHash={item.stakingHash} spendingHash={item.spendingHash} title={item.title} />;
+            }
+          }}
+        />
+      </View>
+
+      {
+        totalPages > 1 && (
+          <View style={{ flexDirection: 'row', gap: 6, marginTop: 12 }}>
+            {
+              circleIndex.map((index) => (
+                <View
+                  key={index + 'indexCard'}
+                  style={{
+                    width: 12,
+                    height: 12,
+                    borderRadius: 100,
+                    backgroundColor: (index - 1) === scrollPosition ? colors.buttonBackgroundBlue : COLORS.GRAY,
+                  }}
+                />
+              ))
+            }
+          </View>
+        )
+      }
+    </>
+  )
+}
+
+export function CaptureShareQRCodeCard({ address }: ShareProps) {
+  const LogoRef = useRef();
+
+  return (
+    <View
+      style={[styles.touchableCard]}
+    >
+      <LinearGradient
+        style={[StyleSheet.absoluteFill, { opacity: 1 }]}
+        start={{ x: 0, y: 0 }}
+        end={{ x: 0, y: 1 }}
+        colors={['#E4E8F7', '#C6F7F7']}
+      />
+
+      <Image ref={LogoRef.current} source={require('../assets/img/yoroi-logo-text.png')} style={{ height: 40, width: 40 }} />
+
+      <View style={styles.addressContainer}>
+        <View style={styles.qrCode}>
+          <QRCode value={address} size={158} backgroundColor={COLORS.LIGHT_GRAY} color="black" />
+        </View>
+
+        <Text style={[styles.textAddress, { color: 'transparent' }]}>{address}</Text>
+      </View>
+    </View>
+  )
 }
 
-export const AddressDetail = ({address}: Props) => (
-  <View style={styles.container}>
-    <QRCode value={address} size={140} backgroundColor={COLORS.LIGHT_GRAY} color="black" />
-  </View>
+export function ShareQRCodeCard({ address, title }: ShareProps) {
+  const [isSharing, setIsSharing] = useState<boolean>(false);
+  const strings = useStrings()
+  const ref = useRef<any>();
+
+  const shareImage = () => {
+    setIsSharing(true);
+
+    setTimeout(async () => {
+      try {
+        const uri = await captureRef(ref, {
+          format: 'png',
+          quality: 1,
+          fileName: 'Share address'
+        });
+        setIsSharing(false);
+        await Share.open({ url: uri, filename: 'Share address', message: `${strings.address} ${address}` });
+      } catch (e) {
+        console.log(e);
```

Comment:
Careful with the logs. Is there's anything better to do here? Otherwise remove it when ready to be merged.